hoomd/md/methods.py

```diff
@@ -225,251 +225,79 @@ class npt(_Method):
     A :py:class:`hoomd.compute.thermo` is automatically specified and associated with *group*.
 
     Examples::
-
-        integrate.npt(group=all, kT=1.0, tau=0.5, tauP=1.0, P=2.0)
-        integrator = integrate.npt(group=all, tau=1.0, kT=0.65, tauP = 1.2, P=2.0)
+        all = filter.All()
+        integrate.NPT(filter=all, kT=1.0, tau=0.5, tauS=1.0, S=2.0)
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0)
         # orthorhombic symmetry
-        integrator = integrate.npt(group=all, tau=1.0, kT=0.65, tauP = 1.2, P=2.0, couple="none")
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0, couple="none")
         # tetragonal symmetry
-        integrator = integrate.npt(group=all, tau=1.0, kT=0.65, tauP = 1.2, P=2.0, couple="xy")
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0, couple="xy")
         # triclinic symmetry
-        integrator = integrate.npt(group=all, tau=1.0, kT=0.65, tauP = 1.2, P=2.0, couple="none", rescale_all=True)
+        integrator = integrate.NPT(filter=all, tau=1.0, kT=0.65, tauS = 1.2, S=2.0, couple="none", rescale_all=True)
     """
-    def __init__(self, group, kT=None, tau=None, S=None, P=None, tauP=None, couple="xyz", x=True, y=True, z=True, xy=False, xz=False, yz=False, all=False, nph=False, rescale_all=None, gamma=None):
-
-        # check the input
-        if (kT is None or tau is None):
-            if nph is False:
-                hoomd.context.current.device.cpp_msg.error("integrate.npt: Need temperature T and thermostat time scale tau.\n")
-                raise RuntimeError("Error setting up NPT integration.")
-            else:
-                # use dummy values
-                kT=1.0
-                tau=1.0
-
-        if (tauP is None):
-                hoomd.context.current.device.cpp_msg.error("integrate.npt: Need barostat time scale tauP.\n")
-                raise RuntimeError("Error setting up NPT integration.")
-
-        # initialize base class
-        _Method.__init__(self)
-
-        # setup the variant inputs
-        kT = hoomd.variant._setup_variant_input(kT)
-
-        # If P is set
-        if P is not None:
-            self.S = [P,P,P,0,0,0]
-        else:
-            # S is a stress, should be [xx, yy, zz, yz, xz, xy]
-            if S is not None and len(S)==6:
-                self.S = S
-            else:
-                raise RuntimeError("Unrecognized stress tensor form")
-
-        S = [hoomd.variant._setup_variant_input(self.S[i]) for i in range(6)]
+    def __init__(self, filter, kT=None, tau=None, S=None, tauS=None, couple="xyz", box_dof=[True,True,True,False,False,False], rescale_all=False, gamma=0.0):
```

Comment:
`kT`, `tau` and others should not default to `None`. I think this was here previously to support the implementation of `NPH`. We should just implement `NPH` with its own param_dict and attach method (or creatively re-use the `NPT` attach method via inheritance).

Reply:
This still needs to be addressed.